lib/fastlane/plugin/wpmreleasetoolkit/models/firebase_test_lab_result.rb

```diff
@@ -0,0 +1,42 @@
+module Fastlane
+  class FirebaseTestLabResult
+    def initialize(log_file_path:)
+      raise "No log file found at path #{log_file_path}" unless File.file? log_file_path
+
+      @path = log_file_path
+    end
+
+    # Scan the log file to for indications that the Test Run failed
+    def success?
+      File.readlines(@path).none? { |line| !line.include? 'Failed' }
+    end
+
+    # Parse the log for the "More details are available..." URL
+    def more_details_url
+      File.readlines(@path)
+          .map { |line| URI.extract(line) }
+          .flatten
+          .compact
+          .filter { |string| string.include? 'matrices' }
+          .first
+    end
```

Comment:
💄  Simpler implementation. Also added a check for the URL's host, to ensure we won't accidentally match false positives of any other red herring URLs containing `matrices` anywhere in their path.
```suggestion
    def more_details_url
      File.readlines(@path)
          .flat_map { |line| URI.extract(line) }
          .find { |url| URI(url).host == 'console.firebase.google.com' && url.include?('/matrices/') }
    end
```